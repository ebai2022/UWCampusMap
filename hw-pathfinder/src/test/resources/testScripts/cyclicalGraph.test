# Test a graph that forms a loop through its edges

CreateGraph g

AddNode g n1
AddNode g n2
AddNode g n3
AddNode g n4

AddEdge g n1 n2 5.0
AddEdge g n2 n3 10.0
AddEdge g n3 n4 5.0
AddEdge g n4 n1 5.0

# Path from a node in a cycle to a different node
FindPath g n1 n4

# Path from a node in a cycle to itself
FindPath g n1 n1