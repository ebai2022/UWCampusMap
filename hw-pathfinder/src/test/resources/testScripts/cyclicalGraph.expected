# Test a graph that forms a loop through its edges

created graph g

added node n1 to g
added node n2 to g
added node n3 to g
added node n4 to g

added edge 5.000 from n1 to n2 in g
added edge 10.000 from n2 to n3 in g
added edge 5.000 from n3 to n4 in g
added edge 5.000 from n4 to n1 in g

# Path from a node in a cycle to a different node
path from n1 to n4:
n1 to n2 with weight 5.000
n2 to n3 with weight 10.000
n3 to n4 with weight 5.000
total cost: 20.000

# Path from a node in a cycle to itself
path from n1 to n1:
total cost: 0.000